test/controllers/SelfAssessmentControllerSpec.scala

```diff
@@ -170,4 +174,136 @@ class SelfAssessmentControllerSpec extends BaseSpec with CurrentTaxYear with Moc
       status(result) shouldBe BAD_REQUEST
     }
   }
+
+  "Calling SelfAssessmentController.viewPayments" should {
+
+    "return 200 and render viewPayments page" when {
+
+      "a user is an Activated SA user" in new LocalSetup {
+
+        override def fakeAuthJourney: FakeAuthJourney =
+          new FakeAuthJourney(ActivatedOnlineFilerSelfAssessmentUser(saUtr))
+
+        val result: Future[Result] = controller.viewPayments()(FakeRequest())
+
+        status(result) shouldBe OK
+
+        Jsoup.parse(contentAsString(result)).text() should include(
+          messagesApi("title.selfAssessment.viewPayments.h1")
+        )
+
+      }
+    }
+
+    "redirect to the home page" when {
+
+      "a user is not an Activated SA user" in new LocalSetup {
+
+        override def fakeAuthJourney: FakeAuthJourney = new FakeAuthJourney(NonFilerSelfAssessmentUser)
+
+        val result =
+          routeWrapper(buildFakeRequestWithAuth("GET", routes.SelfAssessmentController.viewPayments().url))
+            .getOrElse(throw new TestFailedException("Failed to route", 0))
+
+        val doc = Jsoup.parse(contentAsString(result))
+
+        status(result) shouldBe SEE_OTHER
+
+        redirectLocation(result) shouldBe Some(routes.HomeController.index().url)
+      }
+    }
+  }
+
+  "Calling SelfAssessmentController.filterAndSortPayments" should {
+
+    "return an empty list if no payments are present" in new LocalSetup {
+
+      override def fakeAuthJourney: FakeAuthJourney =
+        new FakeAuthJourney(ActivatedOnlineFilerSelfAssessmentUser(saUtr))
+
+      implicit val localDateOrdering: Ordering[LocalDate] = Ordering.fromLessThan(_ isAfter _)
+
+      val list = Some(PaymentSearchResult("PTA", "111111111", List.empty))
+
+      val result = controller.filterAndSortPayments(list)
+
+      result.isEmpty shouldBe true
+    }
+
+    "filter payments to only include payments in the past 60 days" in new LocalSetup {
+
+      override def fakeAuthJourney: FakeAuthJourney =
+        new FakeAuthJourney(ActivatedOnlineFilerSelfAssessmentUser(saUtr))
+
+      implicit val localDateOrdering: Ordering[LocalDate] = Ordering.fromLessThan(_ isAfter _)
+
+      val outlier = SelfAssessmentPayment(LocalDate.now().minusDays(61), "KT123459", 7.00)
+
+      val payments = List(
+        PayApiPayment("Successful", 14587, "KT123457", LocalDateTime.now().minusDays(11.toLong)),
```

Comment:
Is this a Nino?

Reply:
No, it's just a payment reference string that pay-api generates